src/OneSignal.ts

```diff
@@ -92,32 +92,52 @@ export default class OneSignal {
    * @PublicApi
    */
   static async setEmail(email: string, options?: SetEmailOptions): Promise<string> {
-    if (!email)
+    if (!email) {
       throw new InvalidArgumentError('email', InvalidArgumentReason.Empty);
-    if (!isValidEmail(email))
+    }
+    if (!isValidEmail(email)) {
       throw new InvalidArgumentError('email', InvalidArgumentReason.Malformed);
-    // emailAuthHash is expected to be a 64 character SHA-256 hex hash
-    if (options && options.emailAuthHash && options.emailAuthHash.length !== 64) {
-      throw new InvalidArgumentError('options.emailAuthHash', InvalidArgumentReason.Malformed);
     }
-    await awaitOneSignalInitAndSupported();
+
+    const isIdentifierAuthHashDefined = options && !!options.identifierAuthHash;
+    const isEmailAuthHashDefined      = options && !!options.emailAuthHash;
+
+    const authHash = isIdentifierAuthHashDefined ? options.identifierAuthHash :
+    (isEmailAuthHashDefined ? options.emailAuthHash : undefined);
```

Comment:
For backwards compatibility, `emailAuthHash` should be acceptable as well as newer `identifierAuthHash` standard

Reply:
Note: typescript warning thinks `emailAuthHash` may not be defined (discussed with Josh)